### PR TITLE
Birthday events are generated on upgrade

### LINKED ISSUE #22971: Contact birthdays of untouched contacts not displayed in calendar
### Steps to reproduce
1. Install owncloud 9.0
2. Go to the calendar app
3. Enable contact_birthdays calendar

### Expected behaviour
Display all contact's birthday entries

### Actual behaviour
No birthdays are displayed.

When I edit the birthday of a contact, it finally gets displayed. But already existing entries are not displayed until I edit the birthday.

### Server configuration
**Operating system**: Ubuntu 15.10

**Web server:** Nginx

**Database:** Mariadb

**PHP version:** 5.6.11-1ubuntu3.1

**ownCloud version:** (see ownCloud admin page) ownCloud 9.0.0 (stable)

**Calendar version:** (see ownCloud apps page) 1.0

**Updated from an older ownCloud or fresh install:** Updated from 8.2

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - documents: 0.12.0
  - encryption: 1.2.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - mail: 0.3.0
  - notes: 2.0.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: 0.8.1
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - user_ldap: 0.8.0
Disabled:
  - external
  - user_external
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/var\/www\/owncloud\/data",
        "dbtype": "mysql",
        "dbname": "owncloud",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "version": "9.0.0.19",
        "installed": true,
        "forcessl": true,
        "maxZipInputSize": 838860800,
        "allowZipDownload": true,
        "theme": "",
        "log_type": "owncloud",
        "loglevel": 1,
        "log_query": false,
        "logdateformat": "Y.m.d H:i:s",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "mail_smtpmode": "php",
        "mail_from_address": "owncloud",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/localhost",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "forceSSLforSubdomains": true,
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0,
            "dbindex": 0
        },
        "logtimezone": "UTC",
        "asset-pipeline.enabled": false,
        "ldapIgnoreNamingRules": false,
        "trashbin_retention_obligation": "auto",
        "maintenance": false,
        "updatechecker": false
    }
}
```

**Are you using external storage, if yes which one:** nope

**Are you using encryption:** nope

**Are you using an external user-backend, if yes which one:** ldap, but currently logged in as as local user

#### LDAP configuration (delete this part if not used)

```
I think not needed for this issue ...
```

### Client configuration
**Browser:** Firefox 46.0a2

**Operating system:** OSX 10.11.4 beta

**CalDAV-clients:** nope

#### ownCloud log (data/owncloud.log)
```
{"reqId":"zxwRD\/jVxD7gb1SDS9HB","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:51:52"}
{"reqId":"TL4ZXI4\/uQIaHUkc+C5E","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:52:52"}
{"reqId":"umhotEbM17mkwnEjk4J0","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:52:54"}
{"reqId":"o00SWpE84drd7noxf685","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:52:56"}
{"reqId":"s16jvukDksj\/zJvmNqBU","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_Calendar' not found at \/var\/www\/owncloud\/apps\/tasks\/service\/tasksservice.php#112","level":3,"time":"2016.03.08 19:02:36"}
```

refs https://github.com/owncloud/calendar/issues/298

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes https://github.com/owncloud/core/issues/22971

we might want to consider backport

### CODE REVIEW

File: apps/dav/appinfo/update.php [hunk 1]
```diff
@@ -23,3 +23,4 @@
 
 $app = new Application();
 $app->setupCron();
+$app->generateBirthdays();
```
Comment (nickvergessen) [inline apps/dav/appinfo/update.php]: On install as well (that is 8.2 to 9.0) ?

Comment (DeepDiver1975) [inline apps/dav/appinfo/update.php]: 8.2.x to 9.0.1+ you mean? makes sense I guess

Comment (nickvergessen) [inline apps/dav/appinfo/update.php]: yes

Comment (DeepDiver1975) [inline apps/dav/appinfo/update.php]: done

File: apps/dav/appinfo/application.php [hunk 2]
```diff
@@ -210,4 +210,19 @@ public function migrateCalendars() {
...
 	}
+
+	public function generateBirthdays() {
+		try {
+			/** @var BirthdayService $migration */
+			$migration = $this->getContainer()->query('BirthdayService');
+			$userManager = $this->getContainer()->getServer()->getUserManager();
+
+			$userManager->callForAllUsers(function($user) use($migration) {
+				/** @var IUser $user */
```
Comment (nickvergessen) [inline apps/dav/appinfo/application.php]: Add as typehint on the method?

**Comment by DeepDiver1975** [comment]
> we might want to consider backport

@karlitschek requesting backport of this one as well - we should automatically trigger birthday calendar creation - THX

**Comment by DeepDiver1975** [comment]
@nickvergessen @blizzz @PVince81 @schiesbn @icewind1991 please review and test - THX

**Comment by karlitschek** [comment]
please backport

**Comment by nickvergessen** [comment]
Tested and works :+1: 

@schiesbn please also review/test :wink:

**Comment by nickvergessen** [comment]
or @blizzz as you reported the issue

**Comment by icewind1991** [comment]
:+1: code looks good

File: apps/dav/appinfo/application.php [hunk 3]
```diff
@@ -210,4 +210,19 @@ public function migrateCalendars() {
...
 		}
 	}
+
+	public function generateBirthdays() {
+		try {
+			/** @var BirthdayService $migration */
+			$migration = $this->getContainer()->query('BirthdayService');
+			$userManager = $this->getContainer()->getServer()->getUserManager();
+
+			$userManager->callForAllUsers(function($user) use($migration) {
```
Comment (blizzz) [inline apps/dav/appinfo/application.php]: Just a heads up, that'll take a while on big instances.

**Pull request merged**